sigmastate/src/test/scala/sigmastate/PrettyPrintErgoTreeSpecification.scala

```diff
@@ -0,0 +1,443 @@
+    val code = "{ (x: Option[Long]) => x.getOrElse(1L) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Option[Long]) => $1.getOrElse(1.toLong) }"
+  }
+
+  property("proveDlog"){
+    val code = "{ (x: GroupElement) => proveDlog(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: GroupElement) => proveDlog($1) }"
+  }
+
+  property("proveDHTuple"){
+    val code = "{ (x: GroupElement) => proveDHTuple(x, x, x, x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: GroupElement) => proveDHTuple($1, $1, $1, $1) }"
+  }
+  
+  property("sigma and"){
+    val code = "{ (x:(SigmaProp, SigmaProp)) => x._1 && x._2 }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: (SigmaProp, SigmaProp)) => $1._1 && $1._2 }"
+  }
+
+  property("sigma or"){
+    val code = "{ (x:(SigmaProp, SigmaProp)) => x._1 || x._2 }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: (SigmaProp, SigmaProp)) => $1._1 || $1._2 }"
+  }
+
+  property("logical or for collections (anyOf)"){
+    val code = "{ (x: Coll[Boolean]) => anyOf(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[Boolean]) => anyOf($1) }"
+  }
+
+  property("logical xor for collections (xorOf)"){
+    val code = "{ (x: Coll[Boolean]) => xorOf(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[Boolean]) => xorOf($1) }"
+  }
+
+  property("logical and for collections (allOr)"){
+    val code = "{ (x: Coll[Boolean]) => allOf(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[Boolean]) => allOf($1) }"
+  }
+
+  property("atLeast"){
+    val code = "{ (x: Coll[SigmaProp]) => atLeast(x.size - 1, x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[SigmaProp]) => atLeast($1.size - 1.toInt, $1) }"
+  }
+
+  property("downcast"){
+    val code = "{ (x: Short) => x.toByte }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Short) => $1.toByte }"
+  }
+  
+  property("longToByteArray"){
+    val code = "{ (x: Long) => longToByteArray(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Long) => longToByteArray($1) }"
+  }
+
+  property("byteArrayToLong"){
+    val code = "{ (x: Coll[Byte]) => byteArrayToLong(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[Byte]) => byteArrayToLong($1) }"
+  }
+
+  property("byteArrayToBigInt"){
+    val code = "{ (x: Coll[Byte]) => byteArrayToBigInt(x) }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: Coll[Byte]) => byteArrayToBigInt($1) }"
+  }
+
+  property("decode point"){
+    val code = "{ (x: GroupElement) => decodePoint(x.getEncoded) == x }"
+    PrettyPrintErgoTree.prettyPrint(compile(code)) shouldBe
+      "{ ($1: GroupElement) => (decodePoint($1.getEncoded)) == ($1) }"
```

Comment:
`($1)` the parens are not necessary around ValUse

Reply:
done